File: api/java/toolgood.textfilter.api/src/main/java/toolgood/textfilter/api/Impl/KeywordProvider.java
package toolgood.textfilter.api.Impl;

import java.util.Hashtable;
import java.util.Map;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import toolgood.textfilter.api.TextFilterConfig;
import toolgood.textfilter.api.Datas.CommonResult;
import toolgood.textfilter.api.Datas.Keywords.KeywordListResult;
import toolgood.textfilter.api.Interfaces.IKeywordProvider;

public class KeywordProvider extends ProviderBase implements IKeywordProvider {

    public KeywordProvider(TextFilterConfig textFilterConfig) {
        super(textFilterConfig);
    }

    private final String GetKeywordListUrl = "/api/get-keyword-list";
    private final String AddKeywordUrl = "/api/add-keyword";
    private final String EditKeywordUrl = "/api/edit-keyword";
    private final String DeleteKeywordUrl = "/api/delete-keyword";

    @Override
    public KeywordListResult GetKeywordList(String text, Integer type, Integer page, Integer pageSize) {
        Map<String, Object> dictionary = new Hashtable<String, Object>();
        dictionary.put("text", text);
        dictionary.put("type", type);
        dictionary.put("page", page);
        dictionary.put("pageSize", pageSize);
        String content = doPost(GetKeywordListUrl, dictionary);

        Gson gson = new GsonBuilder().create();
        KeywordListResult result = gson.fromJson(content, KeywordListResult.class);
        return result;
    }

    @Override
    public CommonResult AddKeyword(String text, int type, String comment) {
        Map<String, Object> dictionary = new Hashtable<String, Object>();
        dictionary.put("text", text);
        dictionary.put("type", type);
        dictionary.put("comment", comment);
        String content = doPost(AddKeywordUrl, dictionary);

        Gson gson = new GsonBuilder().create();
        CommonResult result = gson.fromJson(content, CommonResult.class);
        return result;
    }

    @Override
    public CommonResult EditKeyword(int id, String text, int type, String comment) {
        Map<String, Object> dictionary = new Hashtable<String, Object>();
        dictionary.put("id", id);
        dictionary.put("text", text);
        dictionary.put("type", type);
        dictionary.put("comment", comment);
        String content = doPost(EditKeywordUrl, dictionary);

        Gson gson = new GsonBuilder().create();
        CommonResult result = gson.fromJson(content, CommonResult.class);
        return result;
    }

    @Override
    public CommonResult DeleteKeyword(int id) {
        Map<String, Object> dictionary = new Hashtable<String, Object>();
        dictionary.put("id", id);
        String content = doPost(DeleteKeywordUrl, dictionary);

        Gson gson = new GsonBuilder().create();
        CommonResult result = gson.fromJson(content, CommonResult.class);
        return result;
    }

}

File: api/csharp/ToolGood.TextFilter.Api/Impl/KeywordProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;
using ToolGood.TextFilter.Api.Datas;
using ToolGood.TextFilter.Api.Datas.Keywords;
using ToolGood.TextFilter.Api.Interfaces;

namespace ToolGood.TextFilter.Api.Impl
{
    public class KeywordProvider : ProviderBase, IKeywordProvider
    {
        public KeywordProvider() : base(TextFilterConfig.Instance) { }
        public KeywordProvider(TextFilterConfig textFilterConfig) : base(textFilterConfig) { }

        private const string GetKeywordListUrl = "/api/get-keyword-list";
        private const string AddKeywordUrl = "/api/add-keyword";
        private const string EditKeywordUrl = "/api/edit-keyword";
        private const string DeleteKeywordUrl = "/api/delete-keyword";

        public KeywordListResult GetKeywordList(string text, int? type, int? page, int? pageSize)
        {
            Dictionary<string, object> dict = new Dictionary<string, object>();
            dict["text"] = text;
            dict["type"] = type;
            dict["page"] = page;
            dict["pageSize"] = pageSize;
            var json = JsonConvert.SerializeObject(dict);
            return PostContent<KeywordListResult>(GetKeywordListUrl, json);

        }
        public Task<KeywordListResult> GetKeywordListAsync(string text, int? type, int? page, int? pageSize)
        {
            Dictionary<string, object> dict = new Dictionary<string, object>();
            dict["text"] = text;
            dict["type"] = type;
            dict["page"] = page;
            dict["pageSize"] = pageSize;
            var json = JsonConvert.SerializeObject(dict);
            return PostContentAsync<KeywordListResult>(GetKeywordListUrl, json);
        }


        public CommonResult AddKeyword(string text, int type, string comment)
        {
            Dictionary<string, object> dict = new Dictionary<string, object>();
            dict["text"] = text;
            dict["type"] = type;
            dict["comment"] = comment;
            var json = JsonConvert.SerializeObject(dict);
            return PostContent<CommonResult>(AddKeywordUrl, json);
        }
        public Task<CommonResult> AddKeywordAsync(string text, int type, string comment)
        {
            Dictionary<string, object> dict = new Dictionary<string, object>();
            dict["text"] = text;
            dict["type"] = type;
            dict["comment"] = comment;
            var json = JsonConvert.SerializeObject(dict);
            return PostContentAsync<CommonResult>(AddKeywordUrl, json);
        }


        public CommonResult EditKeyword(int id, string text, int type, string comment)
        {
            Dictionary<string, object> dict = new Dictionary<string, object>();
            dict["id"] = id;
            dict["text"] = text;
            dict["type"] = type;
            dict["comment"] = comment;
            var json = JsonConvert.SerializeObject(dict);
            return PostContent<CommonResult>(EditKeywordUrl, json);
        }
        public Task<CommonResult> EditKeywordAsync(int id, string text, int type, string comment)
        {
            Dictionary<string, object> dict = new Dictionary<string, object>();
            dict["id"] = id;
            dict["text"] = text;
            dict["type"] = type;
            dict["comment"] = comment;
            var json = JsonConvert.SerializeObject(dict);
            return PostContentAsync<CommonResult>(EditKeywordUrl, json);
        }


        public CommonResult DeleteKeyword(int id)
        {
            Dictionary<string, object> dict = new Dictionary<string, object>();
            dict["id"] = id;
            var json = JsonConvert.SerializeObject(dict);
            return PostContent<CommonResult>(DeleteKeywordUrl, json);
        }
        public Task<CommonResult> DeleteKeywordAsync(int id)
        {
            Dictionary<string, object> dict = new Dictionary<string, object>();
            dict["id"] = id;
            var json = JsonConvert.SerializeObject(dict);
            return PostContentAsync<CommonResult>(DeleteKeywordUrl, json);
        }


    }
}
